store/sqlstore/thread_store.go

```diff
@@ -188,3 +188,47 @@ func (s *SqlThreadStore) CreateMembershipIfNeeded(userId, postId string) error {
 	})
 	return err
 }
+
+func (s *SqlThreadStore) UpdateUnreadsByChannel(userId string, channelLastUnreads map[string]int64) error {
+	var channelIds []string
+	for k := range channelLastUnreads {
+		channelIds = append(channelIds, k)
+	}
+	var threadData []struct {
+		PostId      string
+		ChannelId   string
+		LastReplyAt int64
+	}
+	query, args, _ := s.getQueryBuilder().
+		Select("PostId, ChannelId, LastReplyAt").
+		From("Threads").
+		LeftJoin("Posts ON Threads.PostId=Posts.Id").
+		Where(sq.Eq{"Posts.ChannelId": channelIds}).
```

Comment:
Can we filter here by the current LastUpdated to avoid unnecesary writes? I mean, for example, something like, if the `LastReplyAt < Min(LastReplyAt of channels)`

Reply:
@jespino Sounds good, will do